lib/core/utils/get-ancestry.js

```diff
@@ -0,0 +1,31 @@
+import getShadowSelector from './get-shadow-selector';
+
+function generateAncestry(node) {
+	const nodeName = node.nodeName.toLowerCase();
+	const parent = node.parentElement;
+	if (!parent) {
+		return nodeName;
+	}
+
+	let nthChild = '';
+	if (
+		nodeName !== 'head' &&
+		nodeName !== 'body' &&
+		parent.children.length > 1
```

Comment:
Would it be better if this was only checking if the parent had multiple of the nodeName? I know `getSelector` only adds nth-child stuff if it needs to.

Right now if the parent has two children but they are both different nodeNames, you still get the nth-child selector which doesn't do much but add noise:

```
<html>
  <body>
    <div>
      <a href="/foo">Foo</a>
      <span>Hello</span>
    </div> 
  </body>
</html>
```

Results in `html > body > div:nth-child(1) > a:nth-child(1)`

Reply:
What would the benefit of that be? This seems faster.